### PR TITLE
Migrate Gradle files to use aep-plugin library

### PR DESCRIPTION
**Pull request by kevinlind**
## Description

Updates the Edge Gradle files, migrating from Groovy to Kotlin versions, and use the aep-plugin library.
Updates the CircleCi configuration's orb and image versions.
Updates the GitHub workflow publish scripts to use JDK 17 plus create the GH tag/version on main publish.
Adds GitHub workflow version script, same as Core, to update project version from GH Actions.
Removes use of deprecated MobileCore.registerExtension and updates Function test cases which used the API.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ jobs:
     # List of available Android Docker images: https://circleci.com/developer/images/image/cimg/android#image-tags
     executor:
       name: android/android-docker
-      tag: 2022.09.2-node
+      tag: 2024.01.1-node
```
Comment (cacheung) [inline .circleci/config.yml:63]: What's difference between 2024.01.1-node and 2024.01.1?
Other extensions are using 2024.01.1

File: Makefile [hunk 2]
```diff
@@ -65,22 +68,14 @@ assemble-phone-release:
...
-	@echo "Updating version to $(VERSION), Core version to $(CORE-VERSION)"
-	sed -i '' "s/[0-9]*\.[0-9]*\.[0-9]/$(VERSION)/g" ./code/edge/src/main/java/com/adobe/marketing/mobile/EdgeConstants.java
-	sed -i '' "s/\(moduleVersion=\)[0-9]*\.[0-9]*\.[0-9]/\1$(VERSION)/g" ./code/gradle.properties
-	@if [ -z "$(CORE-VERSION)" ]; then \
-		echo "CORE-VERSION was not provided, skipping"; \
-	else \
-		sed -i '' "s/\(mavenCoreVersion=\)[0-9]*\.[0-9]*\.[0-9]/\1$(CORE-VERSION)/g" ./code/gradle.properties; \
-	fi
+ci-publish: clean assemble-phone-release
+	(./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) publishReleasePublicationToSonatypeRepository -Prelease)
```
Comment (cacheung) [inline Makefile:81]: we can remove `assemble-phone-release` for `-publish-staging` and `ci-publish`, they are taken care by `aep-library`.
refer to https://github.com/adobe/aepsdk-edgeconsent-android/pull/77#discussion_r1483768071

File: .circleci/config.yml (hunk 1 above)
Comment (kevinlind) [reply .circleci/config.yml:63]: `-node` includes Node.js in the image, which is required when using Spotless + Prettier.

File: code/app/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        removeUnusedImports()
+        prettier(mapOf("prettier" to "2.7.1", "prettier-plugin-java" to "1.6.2"))
+        .config(mapOf("parser" to "java", "tabWidth" to 4, "useTabs" to true, "printWidth" to 120))
+        endWithNewline()
+        licenseHeader(BuildConstants.ADOBE_LICENSE_HEADER)
+    }
+}
+
+android {
+    namespace = "com.adobe.marketing.tester"
```
Comment (cacheung) [inline code/app/build.gradle.kts:35]: we need to refactor the app name space to unique namespace
com.adobe.marketing.mobile.<extension>.testapp

Comment (kevinlind) [reply code/app/build.gradle.kts:35]: The namespace matches the directory structure, so I'll need to move and update all the test app files. I'll do this in another PR.

**Review by cacheung — APPROVED** [review]

**Pull request merged**